ocaml/xapi/xapi_vm.ml

```diff
@@ -120,9 +120,25 @@ let compute_memory_overhead = compute_memory_overhead
 
 open Xapi_vm_memory_constraints
 
+(* Since dom0 is not started by the toolstack but by Xen at boot time,
+   we have to modify the Xen command line in order to update dom0's
+   memory allocation. *)
+let set_dom0_memory ~__context ~self ~bytes =
+	let arg = Printf.sprintf "dom0_mem=%LdB,max:%LdB" bytes bytes in
+	let args = ["--set-xen"; arg] in
+	try
+		let _ = Helpers.call_script !Xapi_globs.xen_cmdline_script args in
+		Xapi_host_helpers.Host_requires_reboot.set ()
+	with
+	| _ ->
+		raise Api_errors.(Server_error (internal_error, ["Failed to update control domain memory"]))
```

Comment:
It would also be more helpful if the message read dom0 instead of control domain in general